sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -257,12 +282,28 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
    */
   async renewSessionLock(): Promise<Date> {
     this._throwIfReceiverOrConnectionClosed();
-    await this._createMessageSessionIfDoesntExist();
+    const retryOptions = this._sessionReceiverOptions.retryOptions || {};
+    retryOptions.timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
 
-    this._messageSession!.sessionLockedUntilUtc = await this._context.managementClient!.renewSessionLock(
-      this.sessionId!
-    );
-    return this._messageSession!.sessionLockedUntilUtc!;
+    const renewSessionLockOperationPromise = async () => {
+      const timeTakenByCreateSession = await this._createMessageSessionIfDoesntExist(
```

Comment:
Side thought that should be addressed outside of this PR:

I just realized that it might be silly to create the message session at this point...
The method is to renew session lock. The assumption is that we have the lock on a session and that it needs to be renewed. If we don't already have a session in hand, then this method should throw an error or be a no-op.

Reply:
While we are on this topic, we should check the behavior of the `SessionClient` class in the .Net library. Do we create the message session if it doesnt exist as part of each $managment link operation there?

Reply:
Makes sense.. logged #8111